framework/src/timesteppers/FunctionDT.C

```diff
@@ -80,27 +125,32 @@ FunctionDT::computeDT()
 {
   Real local_dt = 0;
 
-  if (_interpolate)
-    local_dt = _time_ipol->sample(_time);
-  else // Find where we are
+  if (_use_function)
+    local_dt = _function->value(_time, Point(0, 0, 0));
+  else
   {
-    unsigned int i = 0;
-    if (MooseUtils::relativeFuzzyGreaterEqual(_time, _time_t.back()))
+    if (_interpolate)
+      local_dt = _time_ipol->sample(_time);
+    else // Find where we are
     {
-      i = _time_t.size();
+      unsigned int i = 0;
+      if (MooseUtils::relativeFuzzyGreaterEqual(_time, _time_t.back()))
+      {
+        i = _time_t.size();
+      }
+      else
+      {
+        for (; i < _time_t.size() - 1; i++)
+          if (MooseUtils::relativeFuzzyLessThan(_time, _time_t[i + 1]))
+            break;
+      }
+
+      // Use the last dt after the end
+      if (i == _time_t.size())
+        local_dt = _time_dt.back();
+      else
+        local_dt = _time_dt[i];
```

Comment:
Note: I realize you are just moving code here, but this whole block of code is just clunky. Seems like it could be rewritten in a single find_if. It works though, no action necessary.

Reply:
I agree it is not the best code, but it will be deleted when the `_time_t` and `_time_dt` parameters are removed.